luigi/worker.py

```diff
@@ -368,11 +369,22 @@ def run(self):
                 logger.info("Worker %s was stopped. Shutting down Keep-Alive thread" % self._worker_id)
                 break
             with fork_lock:
+                response = None
                 try:
-                    self._scheduler.ping(worker=self._worker_id)
+                    response = self._scheduler.ping(worker=self._worker_id)
                 except:  # httplib.BadStatusLine:
                     logger.warning('Failed pinging scheduler')
 
+                # handle messages
+                if response is not None:
+                    for message in response["messages"]:
+                        self._message_callback(message)
+
+
+def message_callback(fn):
```

Comment:
Thanks for thinking about security from day one. I really like this decorator. :)